lib/apartment/adapters/abstract_adapter.rb

```diff
@@ -80,10 +85,12 @@ def drop(tenant)
       #   @param {String} tenant name
       #
       def switch!(tenant = nil)
-        return reset if tenant.nil?
+        run_callbacks :switch do
```

Comment:
I'm ok with it being called twice on `switch`, just wasn't sure if this is what you'd expect.